src/Type/Php/ParseUrlFunctionDynamicReturnTypeExtension.php

```diff
@@ -105,10 +123,14 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return $this->componentTypesPairedConstantsForLowercaseString[$componentType->getValue()] ?? new ConstantBooleanType(false);
 		}
 
+		if ($urlType->isUppercaseString()->yes()) {
+			return $this->componentTypesPairedConstantsForUppercaseString[$componentType->getValue()] ?? new ConstantBooleanType(false);
+		}
+
 		return $this->componentTypesPairedConstants[$componentType->getValue()] ?? new ConstantBooleanType(false);
 	}
 
-	private function createAllComponentsReturnType(bool $urlIsLowercase): Type
+	private function createAllComponentsReturnType(bool $urlIsLowercase, bool $urlIsUppercase): Type
```

Comment:
I don't think you handle the case where both $urlIsLowercase and $urlIsUppercase are true.

should be
```
$returnTypes = [
					new ConstantBooleanType(false),
					new NullType(),
					IntegerRangeType::fromInterval(0, 65535),
					new IntersectionType([new StringType(), new AccessoryLowercaseStringType(), new AccessoryUppercaseStringType()]),
					$this->createComponentsArray(true, true),
				];
```

But I'm unsure such a real case exist for url...

Reply:
Addressed in d0685570d

> But I'm unsure such a real case exist for url...

Same here.